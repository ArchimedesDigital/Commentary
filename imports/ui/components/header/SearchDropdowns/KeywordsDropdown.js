import SearchToolDropdown from '/imports/ui/components/header/SearchToolDropdown'; // eslint-disable-line import/no-absolute-path
import SearchTermButton from '/imports/ui/components/header/SearchTermButton'; // eslint-disable-line import/no-absolute-path
import { isActive, dropdownPropTypes, dropdownDefaultProps } from './helpers';

/*
	BEGIN KeywordsDropdown
*/
const KeywordsDropdown = ({ keywords, searchDropdownOpen, toggleSearchDropdown, toggleSearchTerm, filters }) => (
	<SearchToolDropdown
		name="Keywords"
		open={searchDropdownOpen === 'Keywords'}
		toggle={toggleSearchDropdown}
		disabled={false}
	>
		{keywords.map(keyword => (
			<SearchTermButton
				key={keyword._id}
				toggleSearchTerm={toggleSearchTerm}
				label={keyword.title}
				searchTermKey="keywords"
				value={keyword}
				active={isActive(filters, keyword, 'keywords')}
			/>
		))}
	</SearchToolDropdown>
);
KeywordsDropdown.propTypes = {
	keywords: React.PropTypes.arrayOf(React.PropTypes.shape({
		_id: React.PropTypes.string.isRequired,
		title: React.PropTypes.string.isRequired,
		slug: React.PropTypes.string.isRequired,
	})),
	...dropdownPropTypes,
};
KeywordsDropdown.defaultProps = {
	keywords: [],
	...dropdownDefaultProps,
};
/*
	END KeywordsDropdown
*/

export default KeywordsDropdown;
