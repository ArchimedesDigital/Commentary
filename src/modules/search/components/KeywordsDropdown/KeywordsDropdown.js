import React from 'react';
import PropTypes from 'prop-types';

// components:
import SearchToolDropdown from '../SearchToolDropdown';
import SearchTermButton from '../SearchTermButton';

// helpers:
import { isActive, dropdownPropTypes, dropdownDefaultProps } from '../../lib/helpers';

/*
	BEGIN KeywordsDropdown
*/
const KeywordsDropdown = ({ keywords, searchDropdownOpen, toggleSearchDropdown, toggleSearchTerm, filters }) => (
	<SearchToolDropdown
		name="Words"
		open={searchDropdownOpen === 'Words'}
		toggle={toggleSearchDropdown}
		disabled={false}
	>
		{keywords.map(keyword => (
			<SearchTermButton
				key={keyword._id}
				toggleSearchTerm={toggleSearchTerm}
				label={keyword.title}
				searchTermKey="keywords"
				value={keyword}
				active={isActive(filters, keyword, 'keywords')}
			/>
		))}
	</SearchToolDropdown>
);
KeywordsDropdown.propTypes = {
	keywords: PropTypes.arrayOf(PropTypes.shape({
		_id: PropTypes.string.isRequired,
		title: PropTypes.string.isRequired,
		slug: PropTypes.string.isRequired,
	})),
	...dropdownPropTypes,
};
KeywordsDropdown.defaultProps = {
	keywords: [],
	...dropdownDefaultProps,
};
/*
	END KeywordsDropdown
*/

export default KeywordsDropdown;
