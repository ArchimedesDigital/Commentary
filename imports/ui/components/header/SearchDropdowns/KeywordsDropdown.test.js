import React from 'react';
import renderer from 'react-test-renderer';

// component:
import KeywordsDropdown from './KeyideasDropdown';

describe('KeywordsDropdown', () => {
	it('renders correctly', () => {

		const tree = renderer
			.create(<KeywordsDropdown
				toggleSearchDropdown={() => {}}
				toggleSearchTerm={() => {}}
				toggle={() => {}}
			/>)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});

// TODO Fix error:
// Warning: Failed context type: The context `muiTheme` is marked as required in `FlatButton`, but its value is `undefined`.
// 	in FlatButton (created by SearchToolDropdown)
// in div (created by SearchToolDropdown)
// in SearchToolDropdown (created by KeyideasDropdown)
// in KeyideasDropdown

