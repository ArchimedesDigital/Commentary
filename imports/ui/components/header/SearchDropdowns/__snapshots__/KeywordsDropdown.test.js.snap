// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeywordsDropdown renders correctly 1`] = `
<div
  className="dropdown search-dropdown search-dropdown--KeyIdeas "
>
  <button
    className="search-tool dropdown-toggle "
    disabled={false}
    href={undefined}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    onTouchTap={[Function]}
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
        "backgroundColor": "rgba(0, 0, 0, 0)",
        "border": 10,
        "borderRadius": 2,
        "boxSizing": "border-box",
        "color": "rgba(0, 0, 0, 0.87)",
        "cursor": "pointer",
        "display": "inline-block",
        "fontFamily": "Roboto, sans-serif",
        "fontSize": "inherit",
        "fontWeight": "inherit",
        "height": 36,
        "lineHeight": "36px",
        "margin": 0,
        "minWidth": 88,
        "muiPrepared": true,
        "outline": "none",
        "overflow": "hidden",
        "padding": 0,
        "position": "relative",
        "textAlign": "center",
        "textDecoration": "none",
        "transform": "translate(0, 0)",
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "userSelect": "none",
        "verticalAlign": null,
      }
    }
    tabIndex={0}
    type="button"
  >
    <div
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      <span
        style={
          Object {
            "fontSize": 14,
            "fontWeight": 500,
            "letterSpacing": 0,
            "muiPrepared": true,
            "paddingLeft": 16,
            "paddingRight": 8,
            "position": "relative",
            "textTransform": "uppercase",
            "verticalAlign": "middle",
          }
        }
      >
        Key Ideas
      </span>
      <span
        className="mdi mdi-chevron-down"
        color="rgba(0, 0, 0, 0.87)"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "display": "inline-block",
            "fontSize": 24,
            "marginLeft": 0,
            "marginRight": 12,
            "muiPrepared": true,
            "position": "relative",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "verticalAlign": "middle",
          }
        }
      />
    </div>
  </button>
  <ul
    className="dropdown-menu"
  >
    <div
      className="dropdown-menu-inner"
    />
    <button
      className="close-dropdown"
      disabled={false}
      href={undefined}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOut={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      onTouchTap={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "background": "none",
          "border": 10,
          "boxSizing": "border-box",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": 0,
          "fontWeight": "inherit",
          "height": 48,
          "margin": 0,
          "muiPrepared": true,
          "outline": "none",
          "overflow": "visible",
          "padding": 12,
          "position": "relative",
          "textDecoration": "none",
          "transform": "translate(0, 0)",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "verticalAlign": null,
          "width": 48,
        }
      }
      tabIndex={0}
      type="button"
    >
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <span
          className="mdi mdi-close"
          color="rgba(0, 0, 0, 0.87)"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "inline-block",
              "fontSize": 24,
              "muiPrepared": true,
              "position": "relative",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
            }
          }
        />
      </div>
    </button>
  </ul>
</div>
`;
